examples/plugins/src/file-size.plugin.unit.test.ts

```diff
@@ -0,0 +1,357 @@
+import { unlink } from 'fs/promises';
+import { vol } from 'memfs';
+import { basename, join } from 'path';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { MEMFS_VOLUME } from '@code-pushup/testing-utils';
+import { formatBytes } from '@code-pushup/utils';
+import {
+  assertFileSize,
+  errorMessage,
+  fileSizeIssues,
+  infoMessage,
+  runnerFunction,
+  scoreFilesizeAudit,
+} from './file-size.plugin';
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+const outputDir = MEMFS_VOLUME;
+const projectJson = JSON.stringify(
+  {
+    test: 42,
+    arr: [1, 2, 3],
+    obj: { test: 42 },
+  },
+  null,
+  2,
+);
+const testJs = `
+    const str = 'Hello World'
+    const num = 42;
+    const obj = ${projectJson};
+  `;
+const readmeMd = '# Docs';
+
+const file = 'test.js';
+
+describe('infoMessage', () => {
+  it.each([['index.js'], [join('src', 'index.js')]])(
+    'should return info message',
+    file => {
+      expect(infoMessage(file, 12)).toBe(
+        `File ${basename(file)} is OK. (size: ${formatBytes(12)})`,
+      );
+    },
+  );
+});
+
+describe('errorMessage', () => {
+  it.each([
+    [1, 0],
+    [2, 1],
+  ])('should return error message', (size, budget) => {
+    expect(errorMessage('test.js', size, budget)).toBe(
+      `File ${file} has ${size} B this is ${1} B too big. (budget: ${budget} B)`,
+    );
+  });
+});
+
+describe('assertFileSize', () => {
+  it.each([[-1], [0], [1]])(
```

Comment:
Based on the [`test.each` docs](https://jestjs.io/docs/api#1-testeachtablename-fn-timeout), this can be simplified:
> If you pass in a 1D array of primitives, internally it will be mapped to a table i.e. `[1, 2, 3] -> [[1], [2], [3]]`

```suggestion
  it.each([-1, 0, 1])(
```